### PR TITLE
Add custom directive to be consumed by external scripts

### PR DESCRIPTION
**Pull request by skorulis-ap**
This adds a `// @knit custom("...")` directive which knit simply passes through into the JSON output. This allows external apps to define custom tags that can be consumed by scripts.
The current planned usage is to have something like `// @knit custom("skip_validation")` which will be consumed by an external validate command to provide an escape hatch for edge cases.

I'm not particularly wedded to the name `custom` so if anyone has a suggestion please let me know.

### CODE REVIEW

File: Tests/KnitCodeGenTests/KnitDirectivesTests.swift [hunk 1]
```diff
@@ -128,6 +128,13 @@ final class KnitDirectivesTests: XCTestCase {
         )
     }
 
+    func testCustom() {
+        XCTAssertEqual(
+            try parse("// @knit custom(\"Foo\") custom(\"Bar\")"),
```
Comment (bradfol) [inline Tests/KnitCodeGenTests/KnitDirectivesTests.swift:133]:
```suggestion
            try parse("// @knit tag(\"Foo\") tag(\"Bar\")"),
```
Naming suggestion: "tag"

**Review by bradfol — APPROVED** [review]
What do you think about including the code for the validation logic within Knit as well? That would make it more of a first-class command than a custom passthrough

File: Tests/KnitCodeGenTests/KnitDirectivesTests.swift (hunk 1 above)
Comment (skorulis-ap) [reply Tests/KnitCodeGenTests/KnitDirectivesTests.swift]: That's better than custom. I've updated the PR to use tag

**Comment by skorulis-ap** [comment]
> What do you think about including the code for the validation logic within Knit as well? That would make it more of a first-class command than a custom passthrough

I've been thinking this through and for the first version of validation I want to treat it as a problem specific to a single codebase and then later look at how it can be generalised.
